rust/src/http2/decompression.rs

```diff
@@ -168,13 +168,13 @@ impl HTTP2DecoderHalf {
     pub fn http2_encoding_fromvec(&mut self, input: &Vec<u8>) {
```

Comment:
Clippy will usually complain about `&Vec` in signatures, especially if a `&[u8]` could be used instead.

Reply:
I meant : does clippy complain about running `to_vec` to do a comparison ?

Reply:
> I meant : does clippy complain about running `to_vec` to do a comparison ?

Not that I can tell.

Reply:
Should it not ?
Should we ask someone ?

Reply:
I'm not seeing a lint that would be called out for this, or one of our supressiosn is also suppressing this.  I suppose we could hunt that down and contribute a lint?  But I think looking at the function signature first, and not using `&Vec` would have resulted in a better choice for the comparison to begin with.

Reply:
Here is a minimal reproducer

```
fn main() {
    let a = vec![0x33, 0x34, 0x35];
// bad
    if a == "345".as_bytes().to_vec() {
        println!("Hello, world!");
    }
// good
    if &a == "345".as_bytes() {
        println!("Hello, world!");
    }
}
```